errors/errorHandler.js

```diff
@@ -1,9 +1,45 @@
+const app = require('../index');
+const bodyParser = require('body-parser');
+const articlesRouter = require('../routes/articles')
+const topicsRouter = require('../routes/topics');
 
 
-function errorHandler(err, req, res, next) {
-    const status = err.status || 500;
-    const message = err.message || "Server error";
-    if (err) res.status(500).send({err});
+
+function handle400Errors(err, req, res, next) {
+  
+  if (err.status === 400) {
+    res.status(400)
+    .send({ error : err.message });
+  } else {
+    next(err);
   }
+}
+
+function handle404Errors(err, req, res, next) {
+  
+  if (err.status === 404) {
+  
+    res.status(404).send( { error : err.message } );
+  } else {
+    next(err);
+  }
+}
+
+function postArticleCommentErrors(err, req, res, next) {
+  // Error for inserting a comment to a article_id with a invalid username, eg /api/articles/3/comments
+  if(err.code === '23503')
```

Comment:
This error could happen elsewhere too so try not to be too specific with the error